### PR TITLE
update for rails6

### PR DESCRIPTION
**Pull request by icorson3**
### Description
Update for Rails 6:
- update travis.yml to include support for latest ruby and rails versions
- remove support for ruby < 2.4 and rails < 4.2
- add gemfiles for rails 6
- add test patch for Rails 4.2 and Ruby 2.6

### References
https://zendesk.atlassian.net/browse/WALR-2073
https://github.com/rails/rails/issues/34790

### Risks
**Low**

### CODE REVIEW

File: test/helper.rb [hunk 1]
```diff
@@ -50,6 +50,21 @@ def change_params(change)
 end
 
 class ActionController::TestCase
+  if RUBY_VERSION>='2.6.0'
+    if Rails.version < '5'
+      class ActionController::TestResponse < ActionDispatch::TestResponse
```
Comment (bquorning) [inline test/helper.rb]: Shouldn’t this be `class ::ActionController::TestResponse`? Or better, reopen the `ActionController` module and define `TestCase` and `TestResponse` classes independent of each other.

File: test/helper.rb [hunk 2]
```diff
@@ -50,6 +50,21 @@ def change_params(change)
 end
 
 class ActionController::TestCase
+  if RUBY_VERSION>='2.6.0'
```
Comment (bquorning) [inline test/helper.rb]:
```suggestion
  if RUBY_VERSION >= '2.6.0'
```

File: .travis.yml [hunk 3]
```diff
@@ -1,17 +1,23 @@
 cache: bundler
 sudo: false
 rvm:
-  - 2.2
-  - 2.3
```
Comment (bquorning) [inline .travis.yml]: We should probably add `required_ruby_version ">= 2.4"` to the gemspec.

File: .travis.yml [hunk 4]
```diff
@@ -1,17 +1,23 @@
...
   - gemfiles/rails4.2.gemfile
   - gemfiles/rails5.0.gemfile
   - gemfiles/rails5.1.gemfile
   - gemfiles/rails5.2.gemfile
+  - gemfiles/rails6.0.gemfile
 script: "bundle exec rake test"
 matrix:
   exclude:
     - rvm: 2.4
-      gemfile: gemfiles/rails3.2.gemfile
```
Comment (bquorning) [inline .travis.yml]: Let’s remove this gemfile from the repository, too.

File: test/helper.rb (hunk 2 above)
Comment (zdennis) [reply test/helper.rb]: This has been cleaned up in d307af4

File: test/helper.rb (hunk 1 above)
Comment (zdennis) [reply test/helper.rb]: We cleaned this up in d307af4. Can you take another look?

File: .travis.yml (hunk 4 above)
Comment (zdennis) [reply .travis.yml]: 👍 removed in 42e4dfe6e0eca2bea48207d2d1361ccf92b2662f

File: .travis.yml (hunk 3 above)
Comment (zdennis) [reply .travis.yml]: Good call. This has been added in 317e33abadf6383be8f2a5a24f114eed2fc7057a .

**Comment by zdennis** [comment]
@bquorning, thanks for reviewing this PR. We've pushed up changes to incorporate your feedback.  Can you take another look?

**Pull request merged**